docs/promise-chaining.md

```diff
@@ -0,0 +1,61 @@
+# Promise Chaining
+
+Imagine that we have a sequence of asynchronous tasks that we would love to be done one after another, while passing the result through the promise chain. This can be very useful for something like loading scripts or other resources. Promises provide a couple of simple ways to achieve that.
+
+Lets take the code we executed from the JS file and analyze it step by step:
+
+    new Promise(function(resolve, reject) { 
+    setTimeout(() => resolve(1), 1000);
+Promise is set to a timeout of 1000 ms, which is 1 second.
```

Comment:
Add code in such a way in .md files
```
// Code here
```

Reply:
Updated.

Github API has been rather limiting in the requests I can send through JavaScript console, so I used an open API for a Library. I hope that is okay.